astro/src/content/blog/building-self-hostable-application.mdx

```diff
@@ -0,0 +1,366 @@
+
+Whatever your release strategy, it must be documented for the developers who build on top of your self-hostable product.
+
+### Release Communication
+
+In addition to the nuts and bolts of how a release happens and what impact it has on your users, to make sure they know about it. After all, if they don't, they won't upgrade.
+
+You must communicate to the user that a new release of your package is available. At FusionAuth, we offer:
+
+* an RSS feed
+* a generic new version email list
+* a security focused email list
+* customer outreach when appropriate
+
+One that we don't offer yet is in-app nagging. Depending on how your users interact with your tool, this may be an option.
+
+Plan to spend some time building out appropriate communication mechanisms.
+
+### Performance
+
+Plan to spend time testing performance across multiple environments before every release. This includes:
+
+* the amount of data in the system
+* hardware variations
+* dependency versions
+* major architectural components you support (such as different databases)
+
+### Understanding Product Usage
+
+Ever used tooling which lets you know what features are used by which users? And which users in a new cohort are similar to users who purchased in the past? Such tooling is foundational for operational efficiencies of SaaS vendors everywhere. Again, that tooling is not for you, unless you build it. The more you report, the higher the impact on your users' privacy. But a high level of privacy is one of the valuable parts of self-hosting.
+
+You should capture only what you need to improve the product, be clear about what you are recording, and have a bright line between the usage data and private data stored in your system. For example, record that feature X is being used, but not the details of the data which feature X is operating on.
+
+Usage reporting may be required to charge your users, if that is part of your business model.
+
+Be clear with your customers about what you are gathering and why. Offer them a chance to turn it off. And realize that some self-hosting customers may choose to go beyond turning it off and block network access, which will disable usage statistics reporting.
+
+### Support
+
+Offering great customer support is more difficult with a self-hostable product. You are not only answering questions about how to use your product, some of which may be quite niche or twist your product in different ways. In fact developers often depend on undocumented but observable features; [see Hyrum's Law](https://www.hyrumslaw.com/). On top of that, you also support heterogeneous deployment environments.
```

Comment:
"..twist your product in different ways." You kinda of cover the above comment here.